packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -177,6 +224,10 @@ const RelationModalWrapper = ({ relation, triggerButtonLabel }: RelationModalPro
                         if (modified && !isSubmitting) {
                           setIsConfirmationOpen(true);
                         } else {
+                          // Add current relation to history before opening a new one
+                          if (currentDocumentMeta && Object.keys(currentDocumentMeta).length > 0) {
```

Comment:
Probably we can avoid to check if the currentDocumentMeta length is undefined because it needs to be always defined (https://github.com/strapi/strapi/pull/23074#discussion_r1982983064)
```suggestion
                          if (currentDocumentMeta) {
```